atomix/cluster/src/main/java/io/atomix/raft/cluster/impl/RaftClusterContext.java

```diff
@@ -294,8 +293,10 @@ public void configure(final Configuration configuration) {
       return;
     }
 
+    final var initialType = localMember.getType();
     updateConfiguration(configuration);
-    if (wasPromoted(currentConfig, configuration, localMember)) {
+    final var newType = localMember.getType();
+    if (initialType.ordinal() < newType.ordinal()) {
```

Comment:
@oleschoenburg I did not fully understand this fix? Could you please add a description to the commit or PR?

Reply:
:+1: I've tried to add a helpful commit message